### ISSUE TITLE
Update prompt only appears after certain task

### ISSUE THREAD

**Issue by John4266** [TcNobo/TcNo-Acc-Switcher#142]
Description: For some reason, the "update available" bar prompt
This:
![](https://i.imgur.com/v0HflgL.png)
only appears after opening and closing the Settings menu.

it's already been like that for several versions, i am 100% sure it applies to the version BEFORE (5.0) as i just noticed it again when updated the program.

```
System info:
OS: Windows 10 Pro 64 Bit
```

**Comment by TcNobo** [comment]
I can only assume this is because of your ping to https://tcno.co. I personally haven't experienced this or heard reports of it...
Try running `ping tcno.co` in CMD, and let me know how long it takes. I have a time of ~3ms so basically nothing as it's hosted in South Africa - where I am.

As for fixing it, there's not much I can do until I found out why it's happening, which I can't unless I can recreate it... Other than that at least you are still getting updates.

**Comment by John4266** [comment]
> 

Here is the ping report:
![](https://i.imgur.com/zYIwc0G.png)

**Comment by John4266** [comment]
@TcNobo Here is a video of the issue:
https://i.imgur.com/lsorhtL.mp4

**Comment by TcNobo** [comment]
Thanks for this valuable info :) I'll see if I can recreate this and get to debugging it :)

**Comment by TcNobo** [comment]
I was able to recreate this on a virtual machine, but after restarting it, it appeared again as normal... I have no idea what could be causing this, so I'll have to sit on it for a while :/

**Comment by John4266** [comment]
> I was able to recreate this on a virtual machine, but after restarting it, it appeared again as normal... I have no idea what could be causing this, so I'll have to sit on it for a while :/

I am assuming it is caused because the function `CheckForUpdate` is run in a separate thread, yet nothing updates the main threads view unless you refresh it.

**Comment by TcNobo** [comment]
Yes, this is true. Happy you thought of this or I would never have 😂 Here's the check on page load... [Only once](https://github.com/TcNobo/TcNo-Acc-Switcher/blob/master/TcNo-Acc-Switcher-Server/Pages/Index.razor#L29-L34).
I'll make the popup happen by invoking javascript and getting it to popup as soon as the check has finished ^

**Comment by John4266** [comment]
Hi. Just letting u know that it worked.

**Comment by TcNobo** [comment]
Awesome. Thanks.
